src/main/io/gps.c

```diff
@@ -688,79 +1079,156 @@ void gpsInitUblox(void)
+
                         ubloxSendNAV5Message(gpsConfig()->gps_ublox_acquire_model);
+
+                        break;
+                    case UBLOX_CFG_ANA:
+                        if (gpsData.unitVersion > UBX_VERSION_M7) {
+                            ubloxSendNav5XMessage();
+                            break;
+                        }
+                        gpsData.state_position++;
+                        break;
+                    case UBLOX_MSG_DISABLE_NMEA:
+                        if (ubloxHasValSetGet()) {
+                            gpsData.state_position++;
+                            break;
+                        }
+                        ubloxDisableNMEAValSet();
+                        gpsData.state_position = UBLOX_MSG_RMC + 1; // skip next entries to UBLOX_MSG_SOL
                         break;
                     case UBLOX_MSG_VGS: //Disable NMEA Messages
-                        ubloxSetMessageRate(0xF0, 0x05, 0); // VGS: Course over ground and Ground speed
+                        ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_VTG, 0); // VGS: Course over ground and Ground speed
                         break;
                     case UBLOX_MSG_GSV:
-                        ubloxSetMessageRate(0xF0, 0x03, 0); // GSV: GNSS Satellites in View
+                        ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_GSV, 0); // GSV: GNSS Satellites in View
                         break;
                     case UBLOX_MSG_GLL:
-                        ubloxSetMessageRate(0xF0, 0x01, 0); // GLL: Latitude and longitude, with time of position fix and status
+                        ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_GLL, 0); // GLL: Latitude and longitude, with time of position fix and status
                         break;
                     case UBLOX_MSG_GGA:
-                        ubloxSetMessageRate(0xF0, 0x00, 0); // GGA: Global positioning system fix data
+                        ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_GGA, 0); // GGA: Global positioning system fix data
                         break;
                     case UBLOX_MSG_GSA:
-                        ubloxSetMessageRate(0xF0, 0x02, 0); // GSA: GNSS DOP and Active Satellites
+                        ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_GSA, 0); // GSA: GNSS DOP and Active Satellites
                         break;
                     case UBLOX_MSG_RMC:
-                        ubloxSetMessageRate(0xF0, 0x04, 0); // RMC: Recommended Minimum data
+                        ubloxSetMessageRate(CLASS_NMEA_STD, MSG_NMEA_RMC, 0); // RMC: Recommended Minimum data
+                        break;
+                    case UBLOX_MSG_SOL:
+                        if (gpsData.unitVersion > UBX_VERSION_M8) {
+                            gpsData.state_position++;
+                            break;
+                        }
+                        ubloxSetMessageRate(CLASS_NAV, MSG_NAV_SOL, 1); // set SOL MSG rate
```

Comment:
If we have an M8 (gpsData.unitVersion == UBX_VERSION_M8), I think we'll get stuck here in the UBLOX_MSG_SOL state.

It will fall through the if statement, set the message rate for the UBX-NAV-SOL message, and then break out of the switch. It won't increment the state to move to UBLAX_MSG_RATE (which has a check for less than M8), so I think it will come back and run this state the same way again.

Since M8 supports both UBX-NAV-SOL and UBX-NAV-PVT, we should skip UBX-NAV-SOL on an M8.

My suggestion is to change this if conditional to:
```if (gpsData.unitVersion => UBX_VERSION_M8) {`
    gpsData.state_position++;
    break;
}``

Reply:
We won't get stuck if we get an ACK for this message (which increments the state).

From other conversations, my understanding was that we weren't going to rely on getting ACKs since reportedly the ACK isn't always sent.

Either way, I'd still suggest the change so we don't enable UBX-NAV-SOL for an M8.